### ISSUE TITLE
Is the init file actually correct?

### ISSUE THREAD

**Issue by jhowardmsft** [Microsoft/opengcs#43]
https://github.com/Microsoft/opengcs/blob/master/kernelconfig/4.11/scripts/init_script#L24-L25

Why not just run gcs in the foreground at this point. Then you don't need a shell....

@gupta-ak @soccerGB @beweedon

**Comment by beweedon** [comment]
My understanding was that the shell was so that porthole could have a shell to connect to. Is that correct?

**Comment by beweedon** [comment]
hcsdiag needs to go through the HCS to connect to the VM. Porthole is useful if the HCS believes the container is in a state where processes can't be launched, even though the utility VM is still running. At least, I've found porthole useful for debugging in situations like that.

**Comment by jhowardmsft** [comment]
Right, but for production, having another process is overhead I don't believe we would actually need

**Comment by beweedon** [comment]
That's fair. Maybe it's just something we want for development, then. I'm not sure we have a way to build a dev build vs a production build, though.

**Comment by gupta-ak** [comment]
Ideally, I would like have a script that generates the initrd in this repository. This way we can change the initrd generation here instead of changing the build server scripts and not worry about artifact origins like in mkfs.ext4 (See #45). Should be pretty straight forward to have a compile flag that creates a dev or prod build

**Comment by rn** [comment]
https://github.com/linuxkit/linuxkit/pull/2202/commits/9cb0b96c508bd66e305647859c8f515cf1f68385#diff-d25a25fa527ef4768a33c7a418162d22 pretty much generates the `initrd` automatically by pulling in the various bits

**Comment by rn** [comment]
to clarify, this doesn't create the `initrd` itself, but the root filesystem. With some `docker save`, `tar`, `cpio` and `gzip` this can create a `initrd`. pretty much what the `moby` tool does
